components/chainhook-sdk/src/observer/http.rs

```diff
@@ -58,116 +79,76 @@ pub async fn handle_new_bitcoin_block(
         match download_and_parse_block_with_retry(&http_client, block_hash, bitcoin_config, ctx)
             .await
         {
-            Ok(block) => Some(block),
+            Ok(block) => block,
             Err(e) => {
-                ctx.try_log(|logger| {
-                    slog::warn!(
-                        logger,
-                        "unable to download_and_parse_block: {}",
-                        e.to_string()
-                    )
-                });
-                None
+                return error_response(
+                    format!("unable to download_and_parse_block: {}", e.to_string()),
+                    ctx,
+                )
             }
         };
-    let Some(block) = block else {
-        ctx.try_log(|logger| {
-            slog::crit!(
-                logger,
-                "Could not download bitcoin block after receiving new_burn_block. Exiting Chainhook observer."
-            )
-        });
-        match background_job_tx.lock() {
-            Ok(tx) => {
-                let _ = tx.send(ObserverCommand::Terminate);
-            }
-            Err(e) => {
-                ctx.try_log(|logger| {
-                    slog::crit!(logger, "Could not shut down event observer: {e}")
-                });
-                std::process::exit(1)
-            }
-        }
-
-        return Json(json!({
-            "status": 500,
-            "result": "unable to retrieve_full_block",
-        }));
-    };
 
     let header = block.get_block_header();
     let block_height = header.block_identifier.index;
     prometheus_monitoring.btc_metrics_block_received(block_height);
     match background_job_tx.lock() {
-        Ok(tx) => {
-            let _ = tx.send(ObserverCommand::ProcessBitcoinBlock(block));
-        }
+        Ok(tx) => match tx.send(ObserverCommand::ProcessBitcoinBlock(block)) {
+            Ok(_) => {}
+            Err(e) => {
+                return error_response(
+                    format!("Unable to send stacks chain event: {}", e.to_string()),
+                    ctx,
+                );
+            }
+        },
         Err(e) => {
-            ctx.try_log(|logger| {
-                slog::warn!(
-                    logger,
-                    "unable to acquire background_job_tx: {}",
-                    e.to_string()
-                )
-            });
-            return Json(json!({
-                "status": 500,
-                "result": "Unable to acquire lock",
-            }));
+            return error_response(
+                format!("unable to acquire background_job_tx: {}", e.to_string()),
+                ctx,
+            );
         }
     };
```

Comment:
It seems that this pattern ignoring success cases can be improved. The empty case can be used as an opportunity to log success, for instance. Also is possible to reduce the nested matches.

Follow a suggestion (_not tested_) to reduce nesting and improve the code readability.

```rust
let bg_job_tx_mutex = background_job_tx.lock()
    .map(|tx| tx.send(ObserverCommand::ProcessBitcoinBlock(block))
    .map_err(|e| format!("Unable to send stacks chain event: {}", e)));

match bg_job_tx_mutex {
    Ok(_) => {
        try_info!(ctx, "Successfully sent ProcessBitcoinBlock command");
    }
    Err(e) => {
        let error_msg = format!("Unable to acquire bg_job_tx mutex: {}", e);
        try_error!(ctx, "{}", error_msg);
        return error_response(error_msg, ctx);
    }
}
```

Reply:
Thanks for the suggestion! I just cleaned these parts of the code. However, I don't think logging the success case is necessary because it would add some noise and not provide value to the logs. Let me know if you disagree though.